app/models/components/course/forums_ability_component.rb

```diff
@@ -52,7 +52,7 @@ def allow_student_reply_unlocked_topics
   end
 
   def allow_student_resolve_own_topics
-    can :resolve, Course::Forum::Topic, creator_id: user.id
+    can [:resolve, :toggle_answer], Course::Forum::Topic, creator_id: user.id
```

Comment:
Actually with `toggle_answer`, should we still allow students to `resolve` their own forum topics, or should we remove that functionality?

The only way I can see that happening is if students answer their own question. However, in those cases, I would expect that the student add their answer (either to the first post or a new post) and mark that post as the `answer`, rather than resolving the topic without flagging the answer.

Reply:
I have hide the resolve in the UI already (there's one commit), do you think we should totally remove it?

Reply:
The view is hidden, but the controller action still exists. For unused code, I would prefer if we remove it, and next time if we need it again, we can re-implement. Otherwise we would have lots of unused code lying around, which I think is not a good thing.

Reply:
OK, that makes sense.  We can always get the deleted code back from git history.